corehq/apps/sso/forms.py

```diff
@@ -546,30 +562,38 @@ def clean_logout_url(self):
         return logout_url
 
     def clean_idp_cert_public(self):
-        idp_cert_public = self.cleaned_data['idp_cert_public']
-        _check_required_when_active(self.cleaned_data['is_active'], idp_cert_public)
-        return idp_cert_public
-
-    def clean_date_idp_cert_expiration(self):
-        date_idp_cert_expiration = self.cleaned_data['date_idp_cert_expiration']
-        _check_required_when_active(self.cleaned_data['is_active'], date_idp_cert_expiration)
-        if date_idp_cert_expiration:
+        idp_cert_file = self.cleaned_data['idp_cert_public']
+        if idp_cert_file:
             try:
-                date_idp_cert_expiration = dateutil.parser.parse(date_idp_cert_expiration)
-            except dateutil.parser.ParserError:
+                cert = certificates.get_certificate_from_file(idp_cert_file)
+                public_key = certificates.get_public_key(cert)
+                date_expiration = certificates.get_expiration_date(cert)
+            except certificates.crypto.Error:
```

Comment:
Would it be worth logging this error somewhere so we can troubleshoot further if people have issues here? `log.exception("bad cert file")` would send it to Sentry, assuming there was a logger named `log`.